From 1eba7c03e8689ae5d96e0f258bedf6ed96b1741f Mon Sep 17 00:00:00 2001
From: Vito Caputo <vito.caputo@coreos.com>
Date: Mon, 16 Mar 2015 13:58:52 -0700
Subject: [PATCH 1/3] nspawn: just disable registration on non-sysd hosts

---
 src/nspawn/nspawn.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/nspawn/nspawn.c b/src/nspawn/nspawn.c
index 0d538c2..f1e5c0d 100644
--- a/src/nspawn/nspawn.c
+++ b/src/nspawn/nspawn.c
@@ -2808,9 +2808,10 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
+        /* we're going to run nspawn on non-systemd hosts, we'll just disable registration when non-sysd */
         if (sd_booted() <= 0) {
-                log_error("Not running on a systemd system.");
-                goto finish;
+                /* TODO(vc): there are other things which won't work on non-systemd hosts, but we don't activate them yet via rkt */
+                arg_register = false;
         }
 
         log_close();
-- 
2.1.4

